src/ClientImpl.cpp

```diff
@@ -513,7 +514,10 @@ void ClientImpl::reconnectEventCallback() {
     event_base_once(m_base, -1, EV_TIMEOUT, reconnectCallback, this, &tv);
 }
 
-void ClientImpl::createPendingConnection(const std::string &hostname, const unsigned short port, int64_t time){
+void ClientImpl::createPendingConnection(const std::string &hostname, const unsigned short port, int64_t time) throw (voltdb::ConnectException) {
+    if (m_connectionAttemptList.size() == 0 ||
+        std::find(m_connectionAttemptList.begin(), m_connectionAttemptList.end(), (hostname + boost::lexical_cast<std::string>(port))) == m_connectionAttemptList.end())
+        throw ConnectException();
```

Comment:
When do you remove from m_connectionAttemptList? Should it be at the end? Do you need a list or can you just use a flag? Does it matter if one host is initiated and other not?